src/Microsoft.Data.SqlClient/tests/UnitTests/IO/TdsWriteStreamTest.cs

```diff
@@ -0,0 +1,297 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Threading;
+using Microsoft.Data.SqlClientX.IO;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.UnitTests.IO
+{
+    /// <summary>
+    /// Unit tests for the TdsWriteStream class.
+    /// </summary>
+    public class TdsWriteStreamTest
```

Comment:
So I'm definitely glad there's a renewed commitment to unit tests, and I'm glad to see that they are already paying off by finding bugs. I'd push a bit on going a little further :)

Organizing tests into Arrange / Act / Assert makes test much more readable.

I tend to go method by method for the tests, and try to go through each scenario that might happen. It does result in duplication in a bunch of cases, but it does help with verifying edge cases are covered. This is one thing I haven't gotten ai to generate completely yet. For instance, I would write a few tests `Write_LessThanBuffer` (writes less bytes than the buffer, verifies no flush happens), `Write_EqualToBuffer` (writes up to end of buffer, verifies no flush happens?), `Write_LargetThanBuffer` (writes entire buffer and then some, verifies flush happens). Likewise with Flush, I'd write `Flush_BufferEmpty`, `Flush_BufferPartial`, `Flush_BufferFull`. 

I won't hold this review since this is definitely much better than no tests. But I'd also be happy to add more tests to enhance it.

Reply:
Yeah, there are lot more things that can be done for UTs
Feel free to build on this, I am going to merge this PR. I am happy that I have been able to take care of critical bug fixes due to UTs.

Then I am going to get my Tds Read stream, since that unblocks Cheena.